tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+     * type name.
+     *
+     * @param type binary type name, e.g., {@code java.util.List}
+     * @return the corresponding JavaParser ResolvedReferenceTypeDeclaration
+     * @throws UnsolvedSymbolException if the type cannot be resolved
+     * @throws UnsupportedOperationException if the type is an array or
+     * primitive type
+     */
+    public static ResolvedReferenceTypeDeclaration getResolvedReferenceTypeDeclaration(String type) throws UnsolvedSymbolException, UnsupportedOperationException {
+        return getResolvedType(type).asReferenceType().getTypeDeclaration().get();
+    }
+
+    private static ResolvedReferenceTypeDeclaration getResolvedReferenceTypeDeclaration(ResolvedType resolvedType) throws UnsupportedOperationException {
+        return resolvedType.asReferenceType().getTypeDeclaration().get();
+    }
+
+    /**
+     * @throws UnsupportedOperationException if the type is an array or a primitive type
+     */
+    private static ResolvedType getResolvedType(String type) throws UnsupportedOperationException {
+        CompilationUnit cu = javaParser.parse(SYNTHETIC_CLASS_SOURCE).getResult().get();
+        BlockStmt syntheticMethodBody = getClassOrInterface(cu, SYNTHETIC_CLASS_NAME).addMethod(SYNTHETIC_METHOD_NAME).getBody().get();
+        syntheticMethodBody.addStatement(type + " type1Var;");
+        return getClassOrInterface(cu, SYNTHETIC_CLASS_NAME)
+                .getMethodsByName(SYNTHETIC_METHOD_NAME).get(0)
+                .getBody().get()
+                .getStatements().getLast().get()
+                .asExpressionStmt().getExpression()
+                .asVariableDeclarationExpr().getVariables().get(0)
+                .resolve().getType();
+    }
+
+    /**
+     * Given a fully qualified class name, returns all methods that can be called on top of that type.
+     * If the type is a regular reference type that can be resolved, retrieves applicable methods. If
+     * the type is a reference type but cannot be resolved (e.g., a generic type), retrieves methods
+     * from java.lang.Object. If the type is an array, retrieves methods from java.lang.Object. If the
+     * type is a primitive, throws an IllegalArgumentException.
+     * @throws IllegalArgumentException if the type is not a reference type or an array
+     */
+    public static Set<MethodUsage> getMethodsOfType(String referenceType) throws IllegalArgumentException {
+        ResolvedType resolvedType = null;
+        ResolvedReferenceTypeDeclaration resolvedReferenceTypeDeclaration = null;
+        Set<MethodUsage> methods = new HashSet<>();
+        boolean useObjectMethods = true;
+        try {
+            resolvedType = getResolvedType(referenceType);
+            resolvedReferenceTypeDeclaration = getResolvedReferenceTypeDeclaration(resolvedType);
+            methods.addAll(resolvedReferenceTypeDeclaration.getAllMethods());
+            // Interfaces do not always inherit from Object
```

Comment:
@AML14

Reply:
I added this comment and subsequent check because JavaParser was throwing an exception when evaluating some common method from Object (e.g., `equals`) on an interface instance. I've also found a couple of references where they state that Interface does not inherit from Object, although they should implicitly declare Object methods:

https://stackoverflow.com/questions/6056124/do-interfaces-inherit-from-object-class-in-java
https://programming.guide/java/do-interfaces-inherit-from-object.html

I can remove the comment or replace it with a more appropriate one, but the subsequent check is necessary to avoid errors arising from JavaParser:

```java
// Interfaces do not always inherit from Object
if (!resolvedReferenceTypeDeclaration.isInterface()) {
    useObjectMethods = false;
}
```